### ISSUE TITLE
Terraform 0.5.0 crash with missing S3 bucket specified in tfstate

### ISSUE THREAD

**Issue by duggan** [hashicorp/terraform#1967]
I think I removed the S3 bucket outside of Terraform because it had some objects in it, and removed it from the tf file, though it was still present in the tfstate file.

On running plan or apply, it crashed. Removing the offending bucket entry from the tfstate file fixed it.

Crash log (starting from the panic) - https://gist.github.com/duggan/316dd6b38e20fbb4a7a5

**Comment by apparentlymart** [comment]
This seems like my bad.

Evidently Terraform shouldn't crash in this case, but I'm not sure what the behavior *should* be if the backing store for the state vanishes while terraform remote state is still active. Looks like the Consul backend just gracefully errors out, so unless someone tells me otherwise I'll plan to make the S3 backend behave that way too.

**Comment by apparentlymart** [comment]
Oh wait... I initially thought this was about storing the state *in* S3, but after reading the log more closely it looks like you're talking about a local state that refers to an S3 bucket resource.

However, I will check to make sure that the S3 remote state backend doesn't crash in the same situation.

**Comment by duggan** [comment]
Your reread is correct, I could have been more explicit.

**Comment by catsby** [comment]
Can you still reproduce this? I'm using Terraform 0.5.1 and I don't get a panic/crash, but I get this error:

```
aws_s3_bucket.bar: Refreshing state... (ID: tf-test-bucket-1)
Error refreshing state: 1 error(s) occurred:

* NoSuchBucket: The specified bucket does not exist
```

I see we're setting the id to `""` but not actually returning in `read`, which we probably should

**Comment by duggan** [comment]
Can confirm this was fixed with 0.5.1!

**Comment by catsby** [comment]
Excellent, thanks for following up! 
Closing for now, sorry for the trouble
